modules/core/src/lifecycle/create-props.js

```diff
@@ -1,139 +1,99 @@
 import log from '../utils/log';
 import {isAsyncIterable} from '../utils/iterable-utils';
 import {parsePropTypes} from './prop-types';
+import {PROP_KEYS} from './constants';
+
+const {COMPONENT, ASYNC_ORIGINAL, ASYNC_RESOLVED, ASYNC_DEFAULTS} = PROP_KEYS;
 
 // Create a property object
 export function createProps() {
   const component = this; // eslint-disable-line
 
   // Get default prop object (a prototype chain for now)
-  const propTypeDefs = getPropsPrototypeAndTypes(component.constructor);
-  const propsPrototype = propTypeDefs.defaultProps;
+  const propsPrototype = getPropsPrototype(component.constructor);
 
   // Create a new prop object with default props object in prototype chain
-  const propsInstance = Object.create(propsPrototype, {
-    // Props need a back pointer to the owning component
-    _component: {
-      enumerable: false,
-      value: component
-    },
-    // The supplied (original) values for those async props that are set to url strings or Promises.
-    // In this case, the actual (i.e. resolved) values are looked up from component.internalState
-    _asyncPropOriginalValues: {
-      enumerable: false,
-      value: {}
-    },
-    // Note: the actual (resolved) values for props that are NOT set to urls or Promises.
-    // in this case the values are served directly from this map
-    _asyncPropResolvedValues: {
-      enumerable: false,
-      value: {}
-    }
-  });
+  const propsInstance = Object.create(propsPrototype);
+
+  // Props need a back pointer to the owning component
+  propsInstance[COMPONENT] = component;
+  // The supplied (original) values for those async props that are set to url strings or Promises.
+  // In this case, the actual (i.e. resolved) values are looked up from component.internalState
+  propsInstance[ASYNC_ORIGINAL] = {};
+  // Note: the actual (resolved) values for props that are NOT set to urls or Promises.
+  // in this case the values are served directly from this map
+  propsInstance[ASYNC_RESOLVED] = {};
 
   // "Copy" all sync props
   for (let i = 0; i < arguments.length; ++i) {
-    Object.assign(propsInstance, arguments[i]);
+    const props = arguments[i];
+    // Do not use Object.assign here to avoid copying Symbols
```

Comment:
Any concern that we might have to worry about this distinction elsewhere?

Reply:
The real concern is that user arguments may overwrite the private fields. At the end of this function we call `Object.freeze` so they will be protected from there.